### PR TITLE
LTP: fixed fallocate01 and fallocate03 testcases

### PR DESCRIPTION
**Pull request by vamsikrishna935**
Modified the tests to use root filesystem.

### CODE REVIEW

File: tests/ltp/patches/fallocate01.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
++ Failure reason:
++ fallocate03    1  TCONF  :  fallocate03.c:251: fallocate system call is not implemented
++ fallocate03    2  TCONF  :  fallocate03.c:251: Remaining cases not appropriate for configuration
```
Comment (davidchisnall) [inline tests/ltp/patches/fallocate01.patch:3]: Why is there a patch of fallocate01 talking about fallocate03 failures?

File: tests/ltp/patches/fallocate03.patch [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
++char *tempdir = "tempdir";
+ char *TCID = "fallocate03";
+ char fname[255];
+ int fd;
+@@ -147,8 +148,8 @@ void cleanup(void)
+        if (close(fd) == -1)
+                tst_resm(TWARN | TERRNO, "close(%s) failed", fname);
+
+-       tst_rmdir();
```
Comment (davidchisnall) [inline tests/ltp/patches/fallocate03.patch:22]: Why is this changed?  We seem to be patching every file that uses `tst_rmdir`, rather than modifying what `tst_rmdir` and friends do in a single place.

**Review by davidchisnall — CHANGES_REQUESTED** [review]
We seem to be patching a lot of tests in identical ways, why are we not centrally fixing the helpers that they are all using?

As before, please track that these centrailly somewhere (and, ideally, on the related issue) so that we can revert the changes when we have fixed the problem that required the test to be patched.  This is not virtual-memory related, so we should eventually be able to make it work.

**Comment by davidchisnall** [comment]
Please re-raise this against the lsds/ltp repo.  Please also address the comments about patching individual tests rather than generic infrastructure.  A lot of tests seem to use the same helpers to use a different filesystem, we should be able to just fix those helpers rather than modifying every single test that relates to them.

**Comment by vamsikrishna935** [comment]
> 
> 
> Please re-raise this against the lsds/ltp repo. Please also address the comments about patching individual tests rather than generic infrastructure. A lot of tests seem to use the same helpers to use a different filesystem, we should be able to just fix those helpers rather than modifying every single test that relates to them.

@davidchisnall  In very few tests, we have changed the temporary directory creation, changing this method will effect existing tests. Could you please let us if we can create another helper method

**Comment by davidchisnall** [comment]
Does changing the generic code cause test failures?  If not, then do that.  If so, add a helper that does what we need for these changes.
